module/batch-action/src/Application/Provider/BatchActionFormProvider.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Application\Provider;
+
+use Ergonode\BatchAction\Application\Form\BatchActionFormInterface;
+use Webmozart\Assert\Assert;
+
+class BatchActionFormProvider
+{
+    /**
+     * @var BatchActionFormInterface[]
+     */
+    private iterable $forms;
+
+    /**
+     * @param BatchActionFormInterface[] $forms
+     */
+    public function __construct(iterable $forms)
+    {
+        Assert::allIsInstanceOf($forms, BatchActionFormInterface::class);
+
+        $this->forms = $forms;
+    }
+
+    public function provide(string $type): ?string
```

Comment:
IMHO its be better move default Form Type hire and always return form class insisted of return null